src/inspsocket.cpp

```diff
@@ -180,42 +180,46 @@ void StreamSocket::DoRead()
 		OnDataReady();
 }
 
-int StreamSocket::ReadToRecvQ(std::string& rq)
+long StreamSocket::ReadToRecvQ(std::string& rq)
 {
-		char* ReadBuffer = ServerInstance->GetReadBuffer();
-		int n = SocketEngine::Recv(this, ReadBuffer, ServerInstance->Config->NetBufferSize, 0);
-		if (n == ServerInstance->Config->NetBufferSize)
+	char* ReadBuffer = ServerInstance->GetReadBuffer();
+	ssize_t n = SocketEngine::Recv(this, ReadBuffer, ServerInstance->Config->NetBufferSize, 0);
+	if (n >= 0)
+	{
+		unsigned long nrecv = static_cast<unsigned long>(n);
+		if (nrecv == ServerInstance->Config->NetBufferSize)
 		{
 			SocketEngine::ChangeEventMask(this, FD_WANT_FAST_READ | FD_ADD_TRIAL_READ);
 			rq.append(ReadBuffer, n);
 		}
-		else if (n > 0)
+		else if (nrecv > 0)
 		{
 			SocketEngine::ChangeEventMask(this, FD_WANT_FAST_READ);
 			rq.append(ReadBuffer, n);
 		}
-		else if (n == 0)
+		else if (nrecv == 0)
 		{
 			error = "Connection closed";
 			SocketEngine::ChangeEventMask(this, FD_WANT_NO_READ | FD_WANT_NO_WRITE);
 			return -1;
 		}
-		else if (SocketEngine::IgnoreError())
-		{
-			SocketEngine::ChangeEventMask(this, FD_WANT_FAST_READ | FD_READ_WILL_BLOCK);
-			return 0;
-		}
-		else if (errno == EINTR)
-		{
-			SocketEngine::ChangeEventMask(this, FD_WANT_FAST_READ | FD_ADD_TRIAL_READ);
-			return 0;
-		}
-		else
-		{
-			error = SocketEngine::LastError();
-			SocketEngine::ChangeEventMask(this, FD_WANT_NO_READ | FD_WANT_NO_WRITE);
-			return -1;
-		}
+	}
```

Comment:
Broken indentation.

Reply:
i think this one is ok. i've added an extra clause before casting away signedness above so there is one more level of indentation expected.

Reply:
Ah, I see!